src/metabase/channel/http.clj

```diff
@@ -0,0 +1,45 @@
+(ns metabase.channel.http
+  (:require
+   [clj-http.client :as http]
+   [metabase.channel.core :as channel]
+   [metabase.channel.shared :as channel.shared]
+   [metabase.util.i18n :refer [tru]]
+   [metabase.util.malli :as mu]
+   [metabase.util.malli.schema :as ms]))
+
+(def ^:private string-or-keyword
+  [:or :string :keyword])
+
+(def ^:private HTTPDetails
+  [:map
+   [:url                           ms/Url]
+   [:auth-method                   [:enum :none :header :query-param :request-body]]
+   [:method       {:optional true} [:enum :get :post :put]]
+   [:query-params {:optional true} [:maybe [:map-of string-or-keyword :any]]]
+   [:headers      {:optional true} [:maybe [:map-of string-or-keyword :any]]]
+   [:body         {:optional true} :string]])
+
+(mu/defmethod channel/send! :channel/http
+  [{:keys [url method auth-method auth-info]} :- HTTPDetails
+   request]
+  (http/request (merge
+                 {:accept       :json
+                  :content-type :json
+                  :method       :post}
+                 {:url    url
+                  :method method}
+                 (cond-> request
+                   (= :request-body auth-method) (update :body merge auth-info)
+                   (= :header auth-method)       (update :headers merge auth-info)
+                   (= :query-param auth-method)  (update :query-params merge auth-info)))))
+
+(defmethod channel/can-connect? :channel/http
+  [_channel-type details]
+  (channel.shared/validate-channel-details HTTPDetails details)
+  (try
+    (channel/send! (merge {:type :channel/http} details) {})
```

Comment:
Would be nice to send a test with some test data actually so they can see what is going on on their end. With the fields we promise to be non-optional.

But I think it should be passed to `can-connect?` somewhere from the top, not generated in there. Optionally we can leave this for later if you want. :)

Reply:
I have a similar concern here: https://metaboat.slack.com/archives/C07318NQ5M2/p1719804756972109?thread_ts=1719018252.065269&cid=C07318NQ5M2

I think it actually makes more sense if we separate the endpoint we used to test the connection vs the endpoint we use to send the payload. but as Arakaki said, it's follow-up work.

Reply:
I don't think we need to do any authentication calls, IMO "put the token in channel config" is perfectly valid.

Github sends this when you just configure webhook: https://docs.github.com/en/webhooks/webhook-events-and-payloads#ping